tally/statgram.go

```diff
@@ -28,6 +29,7 @@ type Sample struct {
 	valueType   SampleType
 	sampleRate  float64
 	stringValue string
+	replace     bool
```

Comment:
On further thought, I think we could do this by having a value type of `GAUGE_REPLACE` and `GAUGE_MODIFY` which could then trigger different codepaths.